### PR TITLE
Windchild's Personnel Improvements

### PR DESCRIPTION
**Pull request by Windchild292**
This implements gender tracking in MegaMek, weighted name generation, and adds an extraData field to Crew. See MekHQ PR https://github.com/MegaMek/mekhq/pull/1533 for more information about the suite of changes this is a part of.

#1649, #1681, #1712, #1716, and #1717 have all been split off of this PR. Any other information can be found in the MekHQ PR.

I would review the files changed, not by commit. This is because I've split off a few parts to be reviewed and merged elsewhere. As for how to go through it, I'd either start off with the numerous files with minimal changes, or the main files that were modified as part of these changes. The aforementioned main files modified are Crew and MULParser.

### CODE REVIEW

File: megamek/src/megamek/common/Crew.java [hunk 1]
```diff
@@ -1173,20 +1252,6 @@ public void setCurrentPilot(int pos) {
         actedThisTurn[pos] = true;
     }
 
-    /**
-     * Set the gunner slot. If a multicrew cockpit uses the same crew member as both pilot and gunner
-     * (i.e. cockpit command console), sets the pilot as well.
-     *
-     * @param pos The slot index to set as gunner.
-     */
-    public void setCurrentGunner(int pos) {
```
Comment (NickAragua) [inline megamek/src/megamek/common/Crew.java]: What happened to all these functions that got removed?

File: megamek/src/megamek/server/ScenarioLoader.java [hunk 2]
```diff
@@ -555,7 +555,7 @@ private Entity parseEntityLine(String s) throws ScenarioLoaderException {
             System.out.println(String.format("Loading %s", ms.getName())); //$NON-NLS-1$
             Entity e = new MechFileParser(ms.getSourceFile(), ms.getEntryName()).getEntity();
             e.setCrew(new Crew(e.getCrew().getCrewType(), parts[1], 1, Integer.parseInt(parts[2]),
-                    Integer.parseInt(parts[3])));
+                    Integer.parseInt(parts[3]), Integer.parseInt(parts[4]), null));
```
Comment (NickAragua) [inline megamek/src/megamek/server/ScenarioLoader.java]: Isn't this going to break scenarios?

File: megamek/src/megamek/common/Crew.java [hunk 3]
```diff
@@ -294,10 +369,28 @@ public String getNickname(int pos) {
...
+        // The randomize return value is used in MekHQ to create new personnel following a battle,
+        // and should not be changed without ensuring it doesn't break on that side
+        if (pos < gender.length) {
+            return gender[pos];
+        } else {
+            return G_RANDOMIZE;
+        }
+    }
+
+    public static int getGenderAsInt(boolean isFemale) {
```
Comment (NickAragua) [inline megamek/src/megamek/common/Crew.java]: What's the deal with this function? Why not just either pass in ints or switch the whole thing to a boolean, why convert between the two?

Comment (Windchild292) [reply megamek/src/megamek/common/Crew.java]: Temporary function to keep functionality while I finished updating names. I'll be removing this in wave 2.

File: megamek/src/megamek/common/Crew.java (hunk 1 above)
Comment (Windchild292) [reply megamek/src/megamek/common/Crew.java]: Unused for years, so I removed them.

File: megamek/src/megamek/server/ScenarioLoader.java (hunk 2 above)
Comment (Windchild292) [reply megamek/src/megamek/server/ScenarioLoader.java]: I think so, this I'll fix immediately.

Comment (Windchild292) [reply megamek/src/megamek/server/ScenarioLoader.java]: Fixed

File: megamek/src/megamek/server/ScenarioLoader.java [hunk 4]
```diff
@@ -554,9 +554,19 @@ private Entity parseEntityLine(String s) throws ScenarioLoaderException {
...
+            if (parts.length > 4) {
+                e.setCrew(new Crew(e.getCrew().getCrewType(), parts[1], 1,
+                        Integer.parseInt(parts[2]), Integer.parseInt(parts[3]),
+                        Integer.parseInt(parts[4]), null));
+            } else {
+                e.setCrew(new Crew(e.getCrew().getCrewType(), parts[1], 1,
+                        Integer.parseInt(parts[2]), Integer.parseInt(parts[3]),
+                        Crew.G_RANDOMIZE, null));
+            }
+            if (parts.length >= 7) {
```
Comment (NickAragua) [inline megamek/src/megamek/server/ScenarioLoader.java]: Think this might be 'off by one' here as well.

**Pull request merged**